flink-streaming-java/src/test/java/org/apache/flink/streaming/api/operators/AbstractStreamOperatorTest.java

```diff
@@ -444,10 +445,11 @@ public void testCustomRawKeyedStateSnapshotAndRestore() throws Exception {
 		final int maxParallelism = 10;
 		final int numSubtasks = 1;
 		final int subtaskIndex = 0;
-		final KeyGroupRange keyGroupRange = KeyGroupRange.of(0, maxParallelism - 1);
+		final List<Integer> keyGroupsToWrite = Arrays.asList(2, 3, 8);
```

Comment:
Maybe we could extract the common logic out as a utility test method and add another test case in name of like `testCustomRawKeyedStateSnapshotAndRestoreWithPartialWrittenKeyGroups`?